src/main/java/com/hivemq/cli/mqtt/SubscribeMqtt5PublishCallback.java

```diff
@@ -36,39 +37,44 @@
     @Nullable private final File publishFile;
     private final boolean printToStdout;
     private final boolean isBase64;
+    private final boolean isJsonOutput;
     private final boolean showTopics;
     private final Mqtt5Client client;
 
     SubscribeMqtt5PublishCallback(final @NotNull Subscribe subscribe, final @NotNull Mqtt5Client client) {
         printToStdout = subscribe.isPrintToSTDOUT();
         publishFile = subscribe.getPublishFile();
         isBase64 = subscribe.isBase64();
+        isJsonOutput = subscribe.isJsonOutput();
         showTopics = subscribe.showTopics();
         this.client  = client;
     }
 
     @Override
     public void accept(final @NotNull Mqtt5Publish mqtt5Publish) {
 
-        PrintWriter fileWriter = null;
-        byte[] payload = mqtt5Publish.getPayloadAsBytes();
-        String payloadMessage = new String(payload);
+        String message;
 
-        if (publishFile != null) { fileWriter = FileUtils.createFileAppender(publishFile); }
-        if (isBase64) { payloadMessage = Base64.toBase64String(payload); }
-
-        if (fileWriter != null) {
-            fileWriter.println(mqtt5Publish.getTopic() + ": " + payloadMessage);
-            fileWriter.flush();
-            fileWriter.close();
+        if (isJsonOutput) {
+            message = new JsonMqttPublish(mqtt5Publish, isBase64).toString();
+        } else if (isBase64) {
+            message = Base64.toBase64String(mqtt5Publish.getPayloadAsBytes());
+        }
+        else {
+            message = new String(mqtt5Publish.getPayloadAsBytes());
         }
```

Comment:
This is the same as in the MQTT 3 callback, should be moved to a PublishFormatter (for example)